client/jetpack-connect/controller.js

```diff
@@ -96,54 +87,40 @@ export default {
 		next();
 	},
 
-	personal( context ) {
-		const analyticsBasePath = '/jetpack/connect/personal',
-			analyticsPageTitle = 'Jetpack Connect Personal';
-
-		analytics.pageView.record( analyticsBasePath, analyticsPageTitle );
-
-		jetpackConnectFirstStep( context, 'personal' );
-	},
-
-	premium( context ) {
-		const analyticsBasePath = '/jetpack/connect/premium',
-			analyticsPageTitle = 'Jetpack Connect Premium';
-
-		analytics.pageView.record( analyticsBasePath, analyticsPageTitle );
-
-		jetpackConnectFirstStep( context, 'premium' );
-	},
-
 	newSite( context ) {
 		analytics.pageView.record( '/jetpack/new', 'Add a new site (Jetpack)' );
 		jetpackNewSiteSelector( context );
 	},
 
-	pro( context ) {
-		const analyticsBasePath = '/jetpack/connect/pro',
-			analyticsPageTitle = 'Jetpack Install Pro';
-
-		analytics.pageView.record( analyticsBasePath, analyticsPageTitle );
-
-		jetpackConnectFirstStep( context, 'pro' );
-	},
-
-	install( context ) {
-		const analyticsBasePath = '/jetpack/connect/install',
-			analyticsPageTitle = 'Jetpack Install';
+	connect( context ) {
+		const {
+			path,
+			pathname,
+			params
+		} = context;
+		const { type = false } = params;
+		const analyticsPageTitle = get( type, analyticsPageTitleByType, 'Jetpack Connect' );
 
-		analytics.pageView.record( analyticsBasePath, analyticsPageTitle );
+		debug( 'entered connect flow with params %o', params );
 
-		jetpackConnectFirstStep( context, 'install' );
-	},
+		analytics.pageView.record( pathname, analyticsPageTitle );
 
-	connect( context ) {
-		const analyticsBasePath = '/jetpack/connect',
-			analyticsPageTitle = 'Jetpack Connect';
+		removeSidebar( context );
```

Comment:
> ... though I'm thinking now that it may be necessary when arriving from a page with a sidebar.

I gave this a run, and if you're navigating from a page with a sidebar you _do_ need this to remove the sidebar.